File: xLiAd.DiagnosticLogCenter.Agent/GuidHolder.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace xLiAd.DiagnosticLogCenter.Agent
{
    public static class GuidHolder
    {
        /// <summary>
        /// 一次请求内，只有本地使用
        /// </summary>
        public static AsyncLocal<Guid> Holder = new AsyncLocal<Guid>();

        /// <summary>
        /// 子请求
        /// </summary>
        public static AsyncLocal<string> ParentHolder = new AsyncLocal<string>();

        /// <summary>
        /// 子请求HTTP
        /// </summary>
        public static AsyncLocal<string> ParentHttpHolder = new AsyncLocal<string>();

        /// <summary>
        /// 一次请求内，所有站点都一致
        /// </summary>
        public static AsyncLocal<string> TraceIdHolder = new AsyncLocal<string>();

        /// <summary>
        /// 一个页面内，所有站点都一致；如果后端生成，则和 TraceId 一致。
        /// </summary>
        public static AsyncLocal<string> PageIdHolder = new AsyncLocal<string>();
    }
}

File: DiagnosticLogCenter-Java/src/main/java/com/xliad/diagnosticlogcenter/agent/GuidHolder.java
package com.xliad.diagnosticlogcenter.agent;

public class GuidHolder {
    private static final ThreadLocal<String> holder = new ThreadLocal<>();
    private static final ThreadLocal<String> parentHolder = new ThreadLocal<>();
    private static final ThreadLocal<String> parentHttpHolder = new ThreadLocal<>();
    private static final ThreadLocal<String> traceIdHolder = new ThreadLocal<>();
    private static final ThreadLocal<String> pageIdHolder = new ThreadLocal<>();

    public static String getHolder() {
        return holder.get();
    }

    public static void setHolder(String guid) {
        holder.set(guid);
    }

    public static void removeHolder() {
        holder.remove();
    }

    public static String getParentHolder() {
        return parentHolder.get();
    }

    public static void setParentHolder(String parentGuid) {
        parentHolder.set(parentGuid);
    }

    public static void removeParentHolder() {
        parentHolder.remove();
    }

    public static String getParentHttpHolder() {
        return parentHttpHolder.get();
    }

    public static void setParentHttpHolder(String parentHttpId) {
        parentHttpHolder.set(parentHttpId);
    }

    public static void removeParentHttpHolder() {
        parentHttpHolder.remove();
    }

    public static String getTraceIdHolder() {
        return traceIdHolder.get();
    }

    public static void setTraceIdHolder(String traceId) {
        traceIdHolder.set(traceId);
    }

    public static void removeTraceIdHolder() {
        traceIdHolder.remove();
    }

    public static String getPageIdHolder() {
        return pageIdHolder.get();
    }

    public static void setPageIdHolder(String pageId) {
        pageIdHolder.set(pageId);
    }

    public static void removePageIdHolder() {
        pageIdHolder.remove();
    }

    public static void clearAll() {
        holder.remove();
        parentHolder.remove();
        parentHttpHolder.remove();
        traceIdHolder.remove();
        pageIdHolder.remove();
    }
}